fineract-provider/src/main/java/org/apache/fineract/accounting/glaccount/domain/TrialBalance.java

```diff
@@ -100,8 +100,9 @@ public boolean equals(Object obj) {
         }
         TrialBalance other = (TrialBalance) obj;
         return Objects.equals(other.officeId, officeId) && Objects.equals(other.glAccountId, glAccountId)
-                && Objects.equals(other.amount, amount) && Objects.equals(other.entryDate, entryDate)
-                && Objects.equals(other.transactionDate, transactionDate) && Objects.equals(other.closingBalance, closingBalance);
+                && Objects.equals(other.amount, amount) && other.entryDate.compareTo(entryDate) == 0 ? Boolean.TRUE
```

Comment:
@percyashu I don't understand this... given that `entryDate` and `transactionDate` are (both) `java.util.Date`, why is `Objects.equals()` not OK? That does not match what https://errorprone.info/bugpattern/UndefinedEquals seems to be about... what am I missing?

Reply:
Error Prone kept complaining java.util.Date equals() doesn't seem to work as expected. This is because from [this](https://docs.oracle.com/javase/6/docs/api/java/util/Date.html#equals(java.lang.Object)) two Date objects are equal if and only if the getTime method returns the same long value for both.